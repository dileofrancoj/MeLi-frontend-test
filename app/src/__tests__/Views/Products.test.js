import React from "react";
import { shallow } from "enzyme";
import ProductsView from "./../../Views/Products";
import { BrowserRouter as Router } from "react-router-dom";
const initialState = {
  id: 1,
  title: "test",
  price: {
    currency: "ARS",
    amount: 2000,
    decimals: 2,
  },
  picture: "test.jpg",
  condition: "usado",
  free_shipping: true,
  address: "capital",
};

describe("<ProductsView /> test", () => {
  const wrapper = shallow(
    <Router>
      <ProductsView {...initialState} />
    </Router>
  );
  test("Should render a ProductsView", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
