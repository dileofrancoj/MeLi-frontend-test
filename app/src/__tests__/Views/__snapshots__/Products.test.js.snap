// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsView /> test Should render a ProductsView 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ProductsView
    address="capital"
    condition="usado"
    free_shipping={true}
    id={1}
    picture="test.jpg"
    price={
      Object {
        "amount": 2000,
        "currency": "ARS",
        "decimals": 2,
      }
    }
    title="test"
  />
</Router>
`;
